member/signin.js

```diff
@@ -0,0 +1,131 @@
+  message.style.display = 'none';
+}
+
+function isValidEmail(email) {
+  let regExp = /^[0-9a-zA-Z]([-_.]?[0-9a-zA-Z])*@[0-9a-zA-Z]([-_.]?[0-9a-zA-Z])*.[a-zA-Z]{2,3}$/i;
+  return regExp.test(email);
+}
+
+function isValidPassword(password) {
+  return password.length >= 8 && /\d/.test(password) && /[a-zA-Z]/.test(password);
+}
+
+function handleEmailValidation() {
+  const emailValue = emailInput.value.trim();
+
+  if (emailValue === '') {
+    displayErrorMessage(emailInput, noneEmailError);
+    hideErrorMessage(emailError);
+    hideErrorMessage(wrongEmail);
+  } else if (!isValidEmail(emailValue)) {
+    displayErrorMessage(emailInput, emailError);
+    hideErrorMessage(noneEmailError);
+    hideErrorMessage(wrongEmail);
+  } else if (emailValue !== 'test@codeit.com') {
+    displayErrorMessage(emailInput, wrongEmail);
+    hideErrorMessage(noneEmailError);
+    hideErrorMessage(emailError);
+  } else {
+    hideErrorMessage(noneEmailError);
+    hideErrorMessage(emailError);
+    hideErrorMessage(wrongEmail);
+  }
+}
+
+function handlePasswordValidation() {
+  const password = passwordInput.value.trim();
+  const emailValue = emailInput.value.trim();
+
+  if(password === ''){
+    displayErrorMessage(passwordInput, nonePw);
+    hideErrorMessage(passwordError);
+    hideErrorMessage(wrongPw)
+  } else if (!isValidPassword(password)) {
+    displayErrorMessage(passwordInput, passwordError);
+    hideErrorMessage(nonePw);
+    hideErrorMessage(wrongPw);
+  } else if(emailValue === 'test@codeit.com' && password !== 'codeit101') {
+    displayErrorMessage(passwordInput, wrongPw);
+    hideErrorMessage(nonePw);
+    hideErrorMessage(passwordError);
+  } else {
+    hideErrorMessage(passwordError);
+    hideErrorMessage(nonePw);
+    hideErrorMessage(wrongPw);
+  }
+}
+
+function isFormValid() {
+  const userEmail = emailInput.value.trim();
+  const userPassword = passwordInput.value.trim();
+  return (
+    userEmail !== '' && isValidEmail(emailInput.value) && emailInput.value === 'test@codeit.com' && 
+    isValidPassword(userPassword) && userPassword === 'codeit101'
+  );
+}
+
+function redirectToFolderPage() {
+  location.href = '../folder.html';
+}
+
+emailInput.addEventListener('blur', function(){
+  handleEmailValidation();
+});
+
+passwordInput.addEventListener('blur', function(){
+  handlePasswordValidation();
+});
+
+loginButton.addEventListener('click', function (event) {
+  event.preventDefault();
+
+  handleEmailValidation();
+  handlePasswordValidation();
+
+  if (isFormValid()) {
+    redirectToFolderPage();
+  }
+});
+
+const togglePwdButton = document.getElementById('togglePwd');
+
+function togglePasswordVisibility() {
+  const passwordInput = document.getElementById('password');
+  const type = passwordInput.type === 'password' ? 'text' : 'password';
+
+  passwordInput.type = type;
+
+  const eyeIconSrc = type === 'password' ? '../img/eye-off.svg' : '../img/eye-on.svg';
+  togglePwdButton.src = eyeIconSrc;
```

Comment:
```suggestion
  const isPasswordType = passwordInput.type === "password"
  
   togglePwdButton.setAttribute(
    "src",
    `../img/eye-${isPasswordType ? "on" : "off"}.svg`
  );
  passwordInput.setAttribute("type", `${isPasswordType ? "text" : "password"}`);
```

중복이 되는 부분은 최대한 하나로 만들고 변하는 부분만 삼항연산자로 관리하는 방법을 적용해주시고
setAttribute라는 메서드를 이용해 속성을 변경 하는 것이 더 가독성 측면에서 좋을 것 같습니다.

물론 지금 작성한 코드 역시 리팩토링을 할 수 있는 코드라고 생각되는데,
어떤 부분을 변경하는 것이 좋을지 직접 고민해 보시는 것도 좋을 것 같습니다!